### DISCUSSION TITLE
How to remove auto segmentation in 3DSlicer

### DISCUSSION THREAD

**Discussion by hw-ju** [Project-MONAI/MONAILabel#1140] [Q&A]
Hi!
I'm using MONAI Label with 3DSlicer. How can I remove bad auto segmentations generated by an immature model in 3DSlicer?  The auto segmentation is too bad to modify upon using interactive annotation, so I want to remove it and annotate it from scratch.

**Comment by tangy5** [comment]
Hi @hw-ju , in the Slicer, MONAI Label's plugin, you could do this:

![1](https://user-images.githubusercontent.com/58751975/201491590-458fdb4b-a5f2-4aa4-be75-516ec0cc5b90.png)

1. If you want to remove a segment class, click "Remove".
2. If you just want to remove predicted mask but not the segment class, right click the segment, such as "liver", then click the last "clear selected segments", the mask will be gone. And you could annotate from scratch.

**Reply by diazandr3s** [reply]
Thanks for asking this, @hw-ju.
You can either do what @tangy5 recommended or clear the segment as I show in this video using right click over the segment: 

https://user-images.githubusercontent.com/11991079/201758080-f8946a45-6de3-4fde-a8b4-08bf5466afa3.mp4

**Reply by hw-ju** [reply]
Thanks @tangy5 @diazandr3s! Both ways work!

**Reply by hw-ju** [reply]
Hi @tangy5 and @diazandr3s! I have a follow-up question.
In a multi-class segmentation, after I run the auto seg using a monailabel pretrained model and do "clear selected segments" for one class, can I use Segment Editor to create the segmentation just for that one class? If yes, how? I tried using `Paint` to put more seeds for just that one class and then run `Initialize` in `Grow-from-seed`, and masks for other classes became much worse.

**Reply by diazandr3s** [reply]
Hi @hw-ju,

This is because the Grow-from-seed functionality uses the other labels as seeds to update the segments - that's why you're getting this result. There should be a way of using Grow-from-seed only on the selected segment and considering the others as background. This is more of a viewer thing.
Have you tried using the 3D paintbrush to update the segment?

**Reply by lassoan** [reply]
>  There should be a way of using Grow-from-seed only on the selected segment and considering the others as background

Yes, you can hide the segments that you don't want to grow seeds from. You can also create a segment to specify as a region within the segments will be grown. It might not be trivial to get all the settings right. If you need more help with this then you can write to the [3D Slicer forum](https://discourse.slicer.org).

**Reply by hw-ju** [reply]
Thanks a lot @diazandr3s @lassoan! Is there a 3DSlicer tutorial / youtube video showing how to do it?
